src/data_availability.rs

```diff
@@ -63,12 +77,39 @@ struct DABusClient {
 }
 }
 
+#[derive(Debug, serde::Deserialize)]
+struct StreamRequest {
+    start_height: u64,
+    peer_ip: String,
+}
+
+/// A peer we are streaming blocks to
+#[derive(Debug)]
+struct BlockStreamPeer {
+    /// Last timestamp we received a ping from the peer.
+    last_ping: u64,
+    /// Sender to stream blocks to the peer
+    sender: SplitSink<Framed<TcpStream, LengthDelimitedCodec>, Bytes>,
```

Comment:
should we apply the usage of SplitSink to the peers too ? Why do we need it here but not on the peers ?

Reply:
Did it to keep my main "task" synchronous, the `sender` is only used by the DA event loop, so I don't need to mutex/arc it. The receiver is only used by the asynchronous ping processes, so I can move it there. Alternative would have been to `Arc<Mutex<>>` the Framed otherwise.

I think the peers don't need this because reading/writing is in the same task.

Reply:
ok good ! got it, thanks !